backtester/data/data_test.go

```diff
@@ -161,52 +146,39 @@ func TestGetStream(t *testing.T) {
 		},
 	}
 	resp, err = b.GetStream()
-	assert.NoError(t, err)
-
-	if len(resp) != 2 {
-		t.Errorf("received '%v' expected '%v'", len(resp), 2)
-	}
+	require.NoError(t, err)
+	assert.Len(t, resp, 2, "GetStream should return 2 items")
 
 	b = nil
 	_, err = b.GetStream()
-	if !errors.Is(err, gctcommon.ErrNilPointer) {
-		t.Errorf("received '%v' expected '%v'", err, gctcommon.ErrNilPointer)
-	}
+	assert.ErrorIs(t, err, gctcommon.ErrNilPointer)
 }
 
 func TestOffset(t *testing.T) {
 	t.Parallel()
 	b := &Base{}
 	o, err := b.Offset()
-	assert.NoError(t, err)
+	require.NoError(t, err)
+	assert.Zero(t, o, "offset should be zero when not set")
 
-	if o != 0 {
-		t.Errorf("received '%v' expected '%v'", o, 0)
-	}
 	b.offset = 1337
 	o, err = b.Offset()
-	assert.NoError(t, err)
+	require.NoError(t, err)
 
-	if o != 1337 {
-		t.Errorf("received '%v' expected '%v'", o, 1337)
-	}
+	assert.Equal(t, int64(1337), o, "offset value should be correct")
 
 	b = nil
 	_, err = b.Offset()
-	if !errors.Is(err, gctcommon.ErrNilPointer) {
-		t.Errorf("received '%v' expected '%v'", err, gctcommon.ErrNilPointer)
-	}
+	assert.ErrorIs(t, err, gctcommon.ErrNilPointer)
 }
 
 func TestSetStream(t *testing.T) {
 	t.Parallel()
 	b := &Base{}
 	err := b.SetStream(nil)
-	assert.NoError(t, err)
+	require.NoError(t, err)
+	assert.Empty(t, b.stream, "SetStream should not error with nil slice and stream should be empty")
```

Comment:
🐒
* Mixed message, in that the assert.Empty message is talking about errors it's not checking for
🚧 🤏 🏴‍☠️  Borderline change request; Flagging for attention 
* Not sure this test asserts what we want.
It implies that `SetStream` will set b.stream to empty, which it won't.
i.e. If `b.stream` was not Empty before we called `SetStream`, this test would fail.
Considered if `SetStream` should init `b.stream` but that feels wrong, to be internally destructive on error.
In conclusion: Recommend removal of test
```suggestion
```